mautic/mautic
# Campaign event: same day execution fix

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes https://github.com/mautic/mautic/issues/13920, https://github.com/mautic/mautic/pull/14089



#### Description:

I create a simple campaign that I want to start every Wednesday at noon:
![164b9c66-92ec-40d3-bfcf-c1dd0b540309](https://github.com/user-attachments/assets/6903a938-e250-4e4b-926a-23ba05d46d13)

If we add a new contact to the campaign on Wednesday at 13:00 we expect the event to be executed right away. It is Wednesday after 12:00. But instead it gets scheduled to the next week:

![f8057bda-4a4f-48f1-9efa-ce6bc0b98026](https://github.com/user-attachments/assets/899fbd50-450f-49b3-9b68-3ab66c5aaa18)


#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Create an empty segment
3. Create a simple campaign that have the empty segment as the source
4. Add a new event adding 5 points. Schedule it to execute only this day of the week and 1 hour before you have now. For example, if it is Friday 14:57, set it to execute only on Fridays at 14:00 so you can see the problem right away and don’t have to wait a few days.
5. Create a new contact ideally with some email address.
6. Go to the contact list view, check the contact and select in the batch actions to add this contact to your campaign that you have created previously.
7. Execute the campaign or wait for the background job to execute it for you. (depends on your situation)
8. Notice that the contact didn’t get the 5 points and instead the point event is scheduled for next week

#### Other areas of Mautic that may be affected by the change:
1. Just Mautic event scheduling


[//]: # ( As applicable: )
#### List of areas covered by the unit and/or functional tests:
1. The tests cover the steps to test above. There are many other tests testing the between time frame or absolute date scheduling. Some I had to update as the execution will be now faster. Those test values were mostly changed when the bug was introduced so I was changing them back.
